Problem installing Version 1.3.1 -> missing pkg-config freetype and libagg

### Bug report
**Bug summary**
I want to install a specific version of matplotlib (1.3.1) in a virtualenv. I created the virtualenv, activated it and ran pip install. Which resulted in the error: _Command "python setup.py egg_info" failed with error code 1_. It appears that freetype and libagg cannot be found by pkg-config. I checked if the libfreetype6 was installed with `ldconfig -p | grep freetype`. Showing that freetype is properly installed. However when I check the pkg-config it appears that freetype is not listed in it, as `pkg-config --list-all | grep freetype` returns no results.

**As a sidenot,** since the upgrade to Ubuntu 18.04 I had problems with libfreetype rendering fonts, and downgraded and marked the package with apt-mark hold, as suggested [here](https://github.com/adobe/brackets/issues/14290), which fixed the font rendering issues. To test if this is the reason for the bug I undid it and upgrade to the most recent version of libfreetype (still resulting in the same error)

**Code for reproduction**
```
virtualenv env
. ./env/bin/activate
pip install matplotlib==1.3.1
```
**Actual outcome**
```
Collecting matplotlib==1.3.1
  Using cached https://files.pythonhosted.org/packages/d4/d0/17f17792a4d50994397052220dbe3ac9850ecbde0297b7572933fa4a5c98/matplotlib-1.3.1.tar.gz
    Complete output from command python setup.py egg_info:
    ============================================================================
    Edit setup.cfg to change the build options
    
    BUILDING MATPLOTLIB
                matplotlib: yes [1.3.1]
                    python: yes [2.7.15rc1 (default, Apr 15 2018, 21:51:34)
                            [GCC 7.3.0]]
                  platform: yes [linux2]
    
    REQUIRED DEPENDENCIES AND EXTENSIONS
                     numpy: yes [version 1.9.0]
                  dateutil: yes [using dateutil version 2.2]
                   tornado: yes [tornado was not found. It is required for the
                            WebAgg backend. pip/easy_install may attempt to
                            install it after matplotlib.]
                 pyparsing: yes [using pyparsing version 2.0.2]
                     pycxx: yes [Couldn't import.  Using local copy.]
                    libagg: yes [pkg-config information for 'libagg' could not
                            be found. Using local copy.]
                  freetype: no  [pkg-config information for 'freetype2' could
                            not be found.]
                       png: yes [version 1.2.54]
    
    OPTIONAL SUBPACKAGES
               sample_data: yes [installing]
                  toolkits: yes [installing]
                     tests: yes [using nose version 1.3.4]
    
    OPTIONAL BACKEND EXTENSIONS
                    macosx: no  [Mac OS-X only]
                    qt4agg: no  [PyQt4 not found]
                   gtk3agg: no  [Requires pygobject to be installed.]
                 gtk3cairo: no  [Requires cairo to be installed.]
                    gtkagg: no  [Requires pygtk]
                     tkagg: no  [The C/C++ header for Tk (tk.h) could not be
                            found.  You may need to install the development
                            package.]
                     wxagg: no  [requires wxPython]
                       gtk: no  [Requires pygtk]
                       agg: yes [installing]
                     cairo: no  [cairo not found]
                 windowing: no  [Microsoft Windows only]
    
    OPTIONAL LATEX DEPENDENCIES
                    dvipng: no
               ghostscript: yes [version 9.25]
                     latex: no
                   pdftops: yes [version 0.62.0]
    
    ============================================================================
                            * The following required packages can not be built:
                            * freetype
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-install-XFiv23/matplotlib/
```

**Matplotlib version**
  * Operating system: Ubuntu 18.04 Bionic
  * Matplotlib version:  1.3.1
  * Python version: Python 2.7.15rc1
  * Other libraries:  pip 18.1 (for python 2.7)

**Solutions tested** (and failed)
 * Upgrading / Reinstalling freetype
 * Different Version of matplotlib (1.4)
 * Upgrade setuptools (pip install --upgrade setuptools)
 * Upgrade pip (python -m pip install --upgrade pip)
 * Added the -Iv argument to pip (pip install -Iv matplotlib==1.3.1)
 * Installed ez_setup (pip install ez_setup)
 * sudo apt-get install freetype*

Thanks for the fast response. I installed the libfreetype6-dev. Now pkg-config --list-all shows an entry for libfreetype6. And it also change the error log when trying to install matplotlib. The new output includes a ton of copying output. The first error I found looks like this:
```
 src/_png.cpp: In member function ‘Py::Object _png_module::write_png(const Py::Tuple&)’:
  src/_png.cpp:243:17: error: ‘npy_PyFile_DupClose’ was not declared in this scope
               if (npy_PyFile_DupClose(py_file, fp)) {
                   ^~~~~~~~~~~~~~~~~~~
  src/_png.cpp:243:17: note: suggested alternative: ‘npy_PyFile_DupClose2’
               if (npy_PyFile_DupClose(py_file, fp)) {
                   ^~~~~~~~~~~~~~~~~~~
                   npy_PyFile_DupClose2
  src/_png.cpp:264:13: error: ‘npy_PyFile_DupClose’ was not declared in this scope
           if (npy_PyFile_DupClose(py_file, fp)) {
               ^~~~~~~~~~~~~~~~~~~
  src/_png.cpp:264:13: note: suggested alternative: ‘npy_PyFile_DupClose2’
           if (npy_PyFile_DupClose(py_file, fp)) {
               ^~~~~~~~~~~~~~~~~~~
               npy_PyFile_DupClose2
  src/_png.cpp: In member function ‘PyObject* _png_module::_read_png(const Py::Object&, bool, int)’:
  src/_png.cpp:329:43: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
       if ((fp = npy_PyFile_Dup(py_file, "rb")))
                                             ^
  src/_png.cpp:577:13: error: ‘npy_PyFile_DupClose’ was not declared in this scope
           if (npy_PyFile_DupClose(py_file, fp)) {
               ^~~~~~~~~~~~~~~~~~~
  src/_png.cpp:577:13: note: suggested alternative: ‘npy_PyFile_DupClose2’
           if (npy_PyFile_DupClose(py_file, fp)) {
               ^~~~~~~~~~~~~~~~~~~
               npy_PyFile_DupClose2
  error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
  
  ----------------------------------------
  Failed building wheel for matplotlib
```
And the second error (and the end of the output) looks like this:
```
                                                      ^
    src/_png.cpp: In member function ‘Py::Object _png_module::write_png(const Py::Tuple&)’:
    src/_png.cpp:243:17: error: ‘npy_PyFile_DupClose’ was not declared in this scope
                 if (npy_PyFile_DupClose(py_file, fp)) {
                     ^~~~~~~~~~~~~~~~~~~
    src/_png.cpp:243:17: note: suggested alternative: ‘npy_PyFile_DupClose2’
                 if (npy_PyFile_DupClose(py_file, fp)) {
                     ^~~~~~~~~~~~~~~~~~~
                     npy_PyFile_DupClose2
    src/_png.cpp:264:13: error: ‘npy_PyFile_DupClose’ was not declared in this scope
             if (npy_PyFile_DupClose(py_file, fp)) {
                 ^~~~~~~~~~~~~~~~~~~
    src/_png.cpp:264:13: note: suggested alternative: ‘npy_PyFile_DupClose2’
             if (npy_PyFile_DupClose(py_file, fp)) {
                 ^~~~~~~~~~~~~~~~~~~
                 npy_PyFile_DupClose2
    src/_png.cpp: In member function ‘PyObject* _png_module::_read_png(const Py::Object&, bool, int)’:
    src/_png.cpp:329:43: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
         if ((fp = npy_PyFile_Dup(py_file, "rb")))
                                               ^
    src/_png.cpp:577:13: error: ‘npy_PyFile_DupClose’ was not declared in this scope
             if (npy_PyFile_DupClose(py_file, fp)) {
                 ^~~~~~~~~~~~~~~~~~~
    src/_png.cpp:577:13: note: suggested alternative: ‘npy_PyFile_DupClose2’
             if (npy_PyFile_DupClose(py_file, fp)) {
                 ^~~~~~~~~~~~~~~~~~~
                 npy_PyFile_DupClose2
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
    
    ----------------------------------------
Command "_/some/path/icensored/_env/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-F2nuP1/matplotlib/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-7pu9aw/install-record.txt --single-version-externally-managed --compile --install-headers _/some/path/icensored/_env/include/site/python2.7/matplotlib" failed with error code 1 in /tmp/pip-install-F2nuP1/matplotlib/


```

--- a/doc/devel/testing.rst
+++ b/doc/devel/testing.rst
@@ -298,3 +298,12 @@ edit this file if you want to add new environments to test (e.g.,
 tests are run. For more info on the ``tox.ini`` file, see the `Tox
 Configuration Specification
 <https://tox.readthedocs.io/en/latest/config.html>`_.
+
+Building old versions of Matplotlib
+-----------------------------------
+
+When running a ``git bisect`` to see which commit introduced a certain bug,
+you may (rarely) need to build very old versions of Matplotlib.  The following
+constraints need to be taken into account:
+
+- Matplotlib 1.3 (or earlier) requires numpy 1.8 (or earlier).

